ui/src/components/data-table/data-table.tsx

```diff
@@ -49,16 +49,14 @@ export function DataTable<TData>({
   renderSubComponent,
   children,
   className,
-  ...props
+  setCurrentPageOptions,
+  setPageSizeOptions,
 }: DataTableProps<TData>) {
   const pagination = table.getState().pagination;
   const totalPages = table.getPageCount();
 
   return (
-    <div
-      className={cn('w-full space-y-2.5 overflow-auto', className)}
-      {...props}
-    >
+    <div className={cn('w-full space-y-2.5 overflow-auto', className)}>
```

Comment:
You should keep the {...props} here and above to allow for all div props to be given to the wrapping div.